microsoft/testsuites/cvm/cvm_attestation_tool.py

```diff
@@ -121,6 +122,151 @@ def _save_attestation_report(self, output: str, log_path: Path) -> None:
             f.write(output)
 
 
+class SnpGuest(Tool):
```

Comment:
Instead of introducing a new class with new name, can we streamline this better? In a way, this class could as well be called "AzureCVMAttestationTests". Is there a way to implement this in a single class (or rename classes with more clear names) or in a pattern where we don't need to add the `if isinstance(node.os, Ubuntu):` condition?

We can e.g. pass node.os to the runc_cvm_attestation function as parameter

Reply:
This class represents the [snpguest](https://github.com/virtee/snpguest) tool and should at least be separate from the tool above. The distro check is to install specific packages since package names may be different across distros. It's generally recommended to do that in Lisa I think

Reply:
But: If we do that check, it sounds like this makes the tool/test specific for Azure Linux, similar to as the previous tool is specific to Ubuntu.

Is there any desire and better way to make this more explicit for both tools?